presto-parser/src/main/java/com/facebook/presto/sql/tree/BinaryLiteral.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.facebook.presto.sql.tree.treeutil.BinaryLiteralUtil;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.util.Optional;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class BinaryLiteral
+        extends Literal
+{
+    private final String value;
+    private final Slice slice;
+
+    public BinaryLiteral(String value)
+    {
+        this(Optional.empty(), value);
+    }
+
+    public BinaryLiteral(Optional<NodeLocation> location, String value)
+    {
+        super(location);
+        requireNonNull(value, "value is null");
+        this.value = value.replaceAll("[^a-fA-F0-9]", "");
+        this.slice = BinaryLiteralUtil.fromHexVarchar(Slices.wrappedBuffer(this.value.getBytes(UTF_8)));
```

Comment:
I don't think there is a particular benefit in turning the string into slice here.